### PR TITLE
Integrate es-server into server module

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Integrates `es-server` into `server`.
Continuation of https://github.com/crate/crate/pull/9934, relates to https://github.com/crate/crate/issues/9888

This will be squashed to a single commit - I kept the individual commits for the review, as I think it's easier to read the changes that way.

## Potential Follow up

- Change `compile` directives in gradle to `implementation` where possible
- Remove `CrateCommonPlugin` and `SQLPlugin` stuff and integrate it into `Node`
- Remove `getSettings` from `RepositoryPlugin` - redundant with `TypeSettings` / `settings` on `Repository.Factory`
- Merge `CrateUnitTest` and `ESTestCase`
- Re-introduce a testing module that exposes all testing dependencies
- Lots more

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/Dummy.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class Dummy {
```
Comment (seut) [inline server/src/test/java/Dummy.java]: Is this actually used?

File: server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java [hunk 2]
```diff
@@ -1444,34 +1441,33 @@ public void testNumericScriptOnAllTypes() {
...
+        execute("insert into tbl (id, x) values (1, 1), (2, 2), (3, 3)");
         refresh();
 
-        execute("select _id, _version from locations where id=2");
-        assertNotNull(response.rows()[0][0]);
-        assertNotNull(response.rows()[0][1]);
+        execute("select _id, _version from tbl where id = 2");
+        assertThat(printedTable(response.rows()), is("2| 1\n"));
 
+        /*
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java:1470]: Should this be removed or re-enabled?

File: server/src/test/java/Dummy.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/test/java/Dummy.java]: Yes, there are some tests that load this class dynamically.

File: server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java:1470]: They should be re-enabled, thanks for catching

**Pull request merged**